extensions/librdkafka/ConsumeKafka.cpp

```diff
@@ -0,0 +1,553 @@
+
+    if (!message_wrapper || RD_KAFKA_RESP_ERR_NO_ERROR != message_wrapper->err) {
+      break;
+    }
+    utils::print_kafka_message(message_wrapper.get(), logger_);
+    // Commit offsets on broker for the provided list of partitions
+    logger_->log_info("Committing offset: %" PRId64 ".", message_wrapper->offset);
+    rd_kafka_commit_message(consumer_.get(), message_wrapper.get(), /* async = */ 0);
+  }
+  logger_->log_info("Done resetting offset manually.");
+}
+
+std::string ConsumeKafka::extract_message(const rd_kafka_message_t* rkmessage) const {
+  if (RD_KAFKA_RESP_ERR_NO_ERROR != rkmessage->err) {
+      throw minifi::Exception(ExceptionType::PROCESSOR_EXCEPTION, "ConsumeKafka: received error message from broker.");
+  }
+  return { reinterpret_cast<char*>(rkmessage->payload), rkmessage->len };
+}
+
+std::vector<std::unique_ptr<rd_kafka_message_t, utils::rd_kafka_message_deleter>> ConsumeKafka::poll_kafka_messages() {
+  std::vector<std::unique_ptr<rd_kafka_message_t, utils::rd_kafka_message_deleter>> messages;
+  messages.reserve(max_poll_records_);
+  const auto start = std::chrono::high_resolution_clock::now();
+  auto elapsed = std::chrono::high_resolution_clock::now() - start;
+  while (messages.size() < max_poll_records_ && elapsed < max_poll_time_milliseconds_) {
+    logger_-> log_debug("Polling for new messages for %d milliseconds...", max_poll_time_milliseconds_.count());
+    rd_kafka_message_t* message = rd_kafka_consumer_poll(consumer_.get(), std::chrono::duration_cast<std::chrono::milliseconds>(max_poll_time_milliseconds_ - elapsed).count());
+    if (!message || RD_KAFKA_RESP_ERR_NO_ERROR != message->err) {
+      break;
+    }
+    utils::print_kafka_message(message, logger_);
+    messages.emplace_back(std::move(message), utils::rd_kafka_message_deleter());
+    elapsed = std::chrono::high_resolution_clock::now() - start;
+  }
+  return messages;
+}
+
+utils::KafkaEncoding ConsumeKafka::key_attr_encoding_attr_to_enum() const {
+  if (utils::StringUtils::equalsIgnoreCase(key_attribute_encoding_, KEY_ATTR_ENCODING_UTF_8)) {
+    return utils::KafkaEncoding::UTF8;
+  }
+  if (utils::StringUtils::equalsIgnoreCase(key_attribute_encoding_, KEY_ATTR_ENCODING_HEX)) {
+    return utils::KafkaEncoding::HEX;
+  }
+  throw minifi::Exception(ExceptionType::PROCESSOR_EXCEPTION, "\"Key Attribute Encoding\" property not recognized.");
+}
+
+utils::KafkaEncoding ConsumeKafka::message_header_encoding_attr_to_enum() const {
+  if (utils::StringUtils::equalsIgnoreCase(message_header_encoding_, MSG_HEADER_ENCODING_UTF_8)) {
+    return utils::KafkaEncoding::UTF8;
+  }
+  if (utils::StringUtils::equalsIgnoreCase(message_header_encoding_, MSG_HEADER_ENCODING_HEX)) {
+    return utils::KafkaEncoding::HEX;
+  }
+  throw minifi::Exception(ExceptionType::PROCESSOR_EXCEPTION, "Key Attribute Encoding property not recognized.");
```

Comment:
typo: should be `Message Header Encoding property`

Reply:
Updated exception text.